src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -819,14 +861,20 @@ def update_network_vpn_gateway(instance, address_prefixes=None, sku=None, vpn_ga
         instance.gateway_type = gateway_type
 
     if enable_bgp is not None:
-        instance.enable_bgp = enable_bgp
+        instance.enable_bgp = enable_bgp.lower() == 'true'
```

Comment:
Up to you, but `lower()` is unnecessary since the arg is bound to a choices list